libcrux-ml-dsa/src/simd/avx2/encoding/commitment.rs

```diff
@@ -1,30 +1,69 @@
 use libcrux_intrinsics::avx2::*;
 
+use minicore::abstractions::{bit::Bit, bitvec::BitVec};
+
+#[hax_lib::ensures(|r| {
+    use hax_lib::*;
+    let r = BitVec::<128>::from(r);
+    let simd_unit = BitVec::<256>::from(*simd_unit);
+    forall(|i: u64|
+        implies(i < 32, r[i] == simd_unit[i / 4 * 32 + i % 4])
+    )
+})]
+#[inline(always)]
+fn serialize_4(simd_unit: &Vec256) -> Vec128 {
+    #[hax_lib::fstar::before(r#"
+    [@@FStar.Tactics.V2.(postprocess_with (fun () -> 
+            let done = alloc false in
+            ctrl_rewrite TopDown (fun _ -> if read done then (false, Skip) else (true, Continue))
+                                 (fun _ -> (fun () -> apply_lemma_rw (`${BitVec::<128>::rewrite_pointwise}); write done true)
+                                           `or_else` trefl);
+            let crate = match cur_module () with | crate::_ -> crate | _ -> fail "Empty module name" in
+            norm [primops; iota; delta_namespace ["Core"; crate; "Libcrux_intrinsics"; "Minicore"; "FStar.FunctionalExtensionality"; "Rust_primitives"]; zeta_full];
+            compute ();
+            norm [primops; iota; delta; zeta_full];
+            trefl ()
+            ))]
+    "#)]
+    #[inline(always)]
+    fn aux(simd_unit: Vec256) -> Vec128 {
```

Comment:
Values should be passed by reference. This style isn't great and may negatively impact performance.
Also, this function needs a proper name that tells the reader what it's doing.

Reply:
👍